src/messaging/rabbitmq/consumer.ts

```diff
@@ -90,15 +96,58 @@ export default class RMQConsumer {
     })
   }
 
-  public async cleanUp(): Promise<void> {
-    this.isConnClosing = true
-    if (this.channel != null) {
-      await this.channel.close()
-      console.log(`[Consumer ${this.name}]: Closed channel successfully`)
-    }
-    if (this.conn !== null) {
-      await this.conn.close()
-      console.log(`[Consumer ${this.name}]: Closed connection successfully`)
+  public async cleanUp(isConnClosing = false): Promise<void> {
+    this.isConnClosing = isConnClosing
+
+    try {
+      // Check if channel exists and is still open
+      if (this.channel?.connection) {
+        try {
+          console.log(`[Consumer ${this.name}]: Nacking all...`)
+          await Promise.race([
+            this.channel.nackAll(true),
```

Comment:
Nacking in-flight messages and re-queing them so that post reconnection, they are received again for processing.

This is still a safety mechanism because while creating channel for consuming messages, ACK is required by default to allow RMQ to delete the messages from the queue once processed and we haven't touched this setting.